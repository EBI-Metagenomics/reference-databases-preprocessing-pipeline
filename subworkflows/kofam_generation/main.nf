include { ADD_METADATA_TO_HMM } from '../../modules/local/add_metadata_to_hmm/main.nf'
include { HMMER_HMMPRESS      } from '../../modules/nf-core/hmmer/hmmpress/main.nf'

workflow KOFAM_GENERATION {
    take:
    ko_hmm_dir
    ko_list

    main:
    ADD_METADATA_TO_HMM(ko_hmm_dir, ko_list)

    ADD_METADATA_TO_HMM.out.kofam_modified
        .map { db_file ->
            [[id: "KOFAM_db"], db_file]
        }
        .set { hmmer_input}

    HMMER_HMMPRESS(hmmer_input)
}