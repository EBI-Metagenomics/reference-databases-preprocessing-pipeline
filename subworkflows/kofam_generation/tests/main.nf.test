nextflow_workflow {

    name "Test workflow KOFAM_GENERATION"
    script "../main.nf"
    workflow "KOFAM_GENERATION"

    test("Run KOFAM_GENERATION subwf on compressed input") {

        when {
            workflow {
                """
                input[0] = file("${moduleDir}/tests/data/profiles.tar.gz", checkIfExists: true)
                input[1] = file("${moduleDir}/tests/data/ko_list.gz", checkIfExists: true)
                """
            }
            params {
                    kofam_version = "2025-01-22"
                    outdir = "${outputDir}"
            }
        }

        then {
            assertAll(
            { assert workflow.success },
            { assert snapshot(workflow.out).match() },
            )
        }
    }

    test("Run KOFAM_GENERATION subwf on uncompressed input") {

        when {
            workflow {
                """
                input[0] = "${moduleDir}/tests/data/profiles"
                input[1] = "${moduleDir}/tests/data/ko_list"
                """
            }
            params {
                    kofam_version = "2025-01-22"
                    outdir = "${outputDir}"
            }
        }

        then {
            assertAll(
            { assert workflow.success },
            { assert snapshot(workflow.out).match() },
            )
        }
    }
}
