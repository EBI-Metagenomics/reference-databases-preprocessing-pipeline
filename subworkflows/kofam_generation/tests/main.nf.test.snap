{
    "Run KOFAM_GENERATION subwf on uncompressed input": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "KOFAM_db"
                        },
                        [
                            "kofam_modified.h3f:md5,e813d5b0c879f6a14dd116e6a5ac9bc4",
                            "kofam_modified.h3i:md5,ea7b9f29dade1fd6982ae10ec9c28ce1",
                            "kofam_modified.h3m:md5,639f42269e97bf96d2d7818977c87ea0",
                            "kofam_modified.h3p:md5,a0e0d29a29562f0656f7f3f0d955c4b3"
                        ]
                    ]
                ],
                "kofam_db": [
                    [
                        {
                            "id": "KOFAM_db"
                        },
                        [
                            "kofam_modified.h3f:md5,e813d5b0c879f6a14dd116e6a5ac9bc4",
                            "kofam_modified.h3i:md5,ea7b9f29dade1fd6982ae10ec9c28ce1",
                            "kofam_modified.h3m:md5,639f42269e97bf96d2d7818977c87ea0",
                            "kofam_modified.h3p:md5,a0e0d29a29562f0656f7f3f0d955c4b3"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-25T15:38:10.288193"
    },
    "Run KOFAM_GENERATION subwf on compressed input": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "KOFAM_db"
                        },
                        [
                            "kofam_modified.h3f:md5,e813d5b0c879f6a14dd116e6a5ac9bc4",
                            "kofam_modified.h3i:md5,ea7b9f29dade1fd6982ae10ec9c28ce1",
                            "kofam_modified.h3m:md5,639f42269e97bf96d2d7818977c87ea0",
                            "kofam_modified.h3p:md5,a0e0d29a29562f0656f7f3f0d955c4b3"
                        ]
                    ]
                ],
                "kofam_db": [
                    [
                        {
                            "id": "KOFAM_db"
                        },
                        [
                            "kofam_modified.h3f:md5,e813d5b0c879f6a14dd116e6a5ac9bc4",
                            "kofam_modified.h3i:md5,ea7b9f29dade1fd6982ae10ec9c28ce1",
                            "kofam_modified.h3m:md5,639f42269e97bf96d2d7818977c87ea0",
                            "kofam_modified.h3p:md5,a0e0d29a29562f0656f7f3f0d955c4b3"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-25T15:38:06.201299"
    }
}